### PR TITLE
feat: @amplitude/eslint-config-typescript

### PR DESCRIPTION
**Pull request by jooohhn**
### Summary

Solves linting 

- Create new `@amplitude/eslint-config-typescript` package
- Fixes `yarn lint`
  - Current project structure didn't work because eslint has a special way of managing [shareable configs](https://eslint.org/docs/developer-guide/shareable-configs) that requires prefixing configs with `eslint-config`
- Caveat: Currently unable to test locally with [lerna](https://github.com/lerna/lerna/issues/2011). There seems to be a problem with linking unpublished local lerna packages with the other packages. 
  - However, we can test manually by running `yarn link` in `packages/eslint-config-typescript`, then `yarn link "@amplitude/eslint-config-typescript && yarn link"` in each package
- TODO in another PR(s)
  - Will need to create another PR once we publish `@amplitude/eslint-config-typescript` to add it to each package
  - Will need to fix linting problems. Probably best to do after current open PRs are merged to avoid merge conflicts

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-Node/blob/master/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:   NO

### CODE REVIEW

**Comment by kelvin-lu** [comment]
@jooohhn the local linking with lerna might be because you're not declaring it as one of the workspaces in the top level `package.json`. *But*, I think it might make sense that this isn't treated as a workspace dir since you can't do other things that we need for validation (testing, building, etc.)

File: packages/eslint-config-typescript/package.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  "main": "index.js",
+  "publishConfig": {
+    "registry": "https://registry.yarnpkg.com"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/amplitude/Amplitude-Node.git"
+  },
+  "scripts": {
+    "test": "echo \"Error: run tests from root\" && exit 1"
```
Comment (kelvin-lu) [inline packages/eslint-config-typescript/package.json:21]: remove this?

**Review by kelvin-lu — APPROVED** [review]
🎉

File: packages/eslint-config-typescript/package.json (hunk 1 above)
Comment (jooohhn) [reply packages/eslint-config-typescript/package.json:21]: Auto-generated by lerna, will remove

**Pull request merged**